components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -501,4 +502,53 @@ private static RealmConfiguration getRealmConfiguration(User user) throws Identi
         return ((org.wso2.carbon.user.core.UserStoreManager)userStoreManager)
                 .getSecondaryUserStoreManager(user.getUserStoreDomain()).getRealmConfiguration();
     }
+
+    public static boolean isAccountStateClaimExisting(String tenantDomain) throws IdentityEventException {
+
+        org.wso2.carbon.user.api.UserRealm userRealm = null;
+        ClaimManager claimManager = null;
+
+        RealmService realmService = IdentityRecoveryServiceDataHolder.getInstance().getRealmService();
+        if (realmService != null) {
+            //Get tenant's user realm.
+            try {
+                int tenantId = realmService.getTenantManager().getTenantId(tenantDomain);
+                userRealm = realmService.getTenantUserRealm(tenantId);
+
+            } catch (UserStoreException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Error while retriving user realm in account disable handler", e);
+                }
+                throw new IdentityEventException("Error while retriving user realm in Utils");
+            }
+        }
+        if (userRealm != null) {
+            //Get claim manager for manipulating attributes.
+            try {
+                claimManager = (ClaimManager) userRealm.getClaimManager();
+            } catch (UserStoreException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Error while retriving claim manager in Utils", e);
+                }
+                throw new IdentityEventException("Error while retriving claim manager in Utils");
```

Comment:
Don't we need to throw the exception?
And full-stop for the error message is missing